### PR TITLE
openclonk: fix runtime, avoid copying libs, and update dependencies

### PR DESCRIPTION
**Pull request by cho-m**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Changes:
- Fix blank screen issue on macOS with patch from head
- Fix build on head which replaced `glew` with `epoxy`.
- Avoid copying libraries into app bundle (existing bottles essentially bundle a copy of dylib).
- Drop `boost` which doesn't seem to be used anymore
- Switch `jpeg` to `jpeg-turbo`.

### CODE REVIEW

**Comment by danielnachun** [comment]
I pushed a patch to fix the build failure on Linux due to a missing include.  We should upstream this assuming it works.  After that I think should be ready to go.

**Comment by cho-m** [comment]
> After that I think should be ready to go.

I previously tried building/running this on my local Docker Linux container with a similar patch and hit some issues at game start up.

Still need to look into error to make sure Linux bottle actually works. I can't recall exact issue, but it might have been having some trouble finding/loading some files.

---

On an unrelated side note, I also tried building v8.0+ (with patches) and found that it built but mouse control was broken on macOS. Not sure how important this is vs. updating to newer version.

Mentioned in https://github.com/openclonk/openclonk/pull/131.

File: Formula/openclonk.rb [hunk 1]
```diff
@@ -22,26 +47,76 @@ class Openclonk < Formula
...
   depends_on "libogg"
   depends_on "libpng"
   depends_on "libvorbis"
+  depends_on "tinyxml"
+
+  uses_from_macos "zlib"
+
+  on_linux do
+    depends_on "pkg-config" => :build
+    depends_on "gcc"
```
Comment (cho-m) [inline Formula/openclonk.rb:76]: Based on GCC plan, we probably don't want to add this and we will be blocked on Linux bottling until we can get `mesa` rebottled (dropping in #106975) with a system GCC (either GCC-5 by disabling some drivers/features or by upgrading system Ubuntu/GCC).

Mesa still claims to build with GCC 4.2.0 but may require some parts to be disabled: https://docs.mesa3d.org/install.html#compiler

**Comment by cho-m** [comment]
Haven't been able to run on Linux container but could be from docker layer.

Since I am not sure of reason, I will avoid blocking merge and get macOS fixes in.

**Review by SMillerDev — APPROVED** [review]

**Pull request closed without merging**